src/Common/Commands.Common.Storage/StorageUtilities.cs

```diff
@@ -26,38 +32,143 @@ public static Uri CreateHttpsEndpoint(string endpointUri)
         {
-            var storageServiceResponse = storageClient.StorageAccounts.Get(accountName);
-            var storageKeysResponse = storageClient.StorageAccounts.GetKeys(accountName);
-
-            Uri fileEndpoint = null;
-            Uri blobEndpoint = null;
-            Uri queueEndpoint = null;
-            Uri tableEndpoint = null;
+            if (!TestMockSupport.RunningMocked)
+            {
+                var storageServiceResponse = storageClient.StorageAccounts.GetProperties(resourceGroupName, accountName);
+                Uri blobEndpoint = storageServiceResponse.StorageAccount.PrimaryEndpoints.Blob;
+                Uri queueEndpoint = storageServiceResponse.StorageAccount.PrimaryEndpoints.Queue;
+                Uri tableEndpoint = storageServiceResponse.StorageAccount.PrimaryEndpoints.Table;
+                return new CloudStorageAccount(
+                    GenerateStorageCredentials(storageClient, resourceGroupName, accountName),
+                    blobEndpoint,
+                    queueEndpoint,
+                    tableEndpoint, null);
+            }
+            else
+            {
+                return new CloudStorageAccount(
+                    new StorageCredentials(accountName,
+                        Convert.ToBase64String(Encoding.UTF8.GetBytes(Guid.NewGuid().ToString()))),
+                    new Uri(string.Format("https://{0}.blob.core.windows.net", accountName)),
+                    new Uri(string.Format("https://{0}.queue.core.windows.net", accountName)),
+                    new Uri(string.Format("https://{0}.table.core.windows.net", accountName)),
+                    null);
+            }
+        }
 
-            if (storageServiceResponse.StorageAccount.Properties.Endpoints.Count >= 4)
+        /// <summary>
+        /// Create a cloud storage account using a service management storage client
+        /// </summary>
+        /// <param name="storageClient">The client to use to get storage account details.</param>
+        /// <param name="accountName">The name of the storage account.</param>
+        /// <returns>A CloudStorageAccount that can be used by windows azure storage library to manipulate objects in the storage account.</returns>
+        public static CloudStorageAccount GenerateCloudStorageAccount(IStorageManagementClient storageClient, string accountName)
+        {
+            if (!TestMockSupport.RunningMocked)
             {
-                fileEndpoint = StorageUtilities.CreateHttpsEndpoint(storageServiceResponse.StorageAccount.Properties.Endpoints[3].ToString());
+                var storageServiceResponse = storageClient.StorageAccounts.Get(accountName);
+
+                Uri fileEndpoint = null;
+                Uri blobEndpoint = null;
+                Uri queueEndpoint = null;
+                Uri tableEndpoint = null;
+
+                if (storageServiceResponse.StorageAccount.Properties.Endpoints.Count >= 4)
```

Comment:
Nit: Would be good to describe what endpoint count of 4 means and what endpoints count of 3 means.